api_client_builder.go

```diff
@@ -47,7 +44,9 @@ func EnableBackingStoreForSerializationWriterFactory(factory s.SerializationWrit
 func enableBackingStoreForSerializationRegistry(registry *s.SerializationWriterFactoryRegistry) {
 	for key, value := range registry.ContentTypeAssociatedFactories {
 		if _, ok := value.(*store.BackingStoreSerializationWriterProxyFactory); !ok {
+			registry.Lock()
 			registry.ContentTypeAssociatedFactories[key] = store.NewBackingStoreSerializationWriterProxyFactory(value)
+			registry.Unlock()
 		}
 	}
 }
```

Comment:
Could we instead use a single lock and unlock for the iteration instead of multiple locks should improve perf

```go
func enableBackingStoreForSerializationRegistry(registry *s.SerializationWriterFactoryRegistry) {
       registry.Lock()
       defer registry.Unlock()
	for key, value := range registry.ContentTypeAssociatedFactories {
		if _, ok := value.(*store.BackingStoreSerializationWriterProxyFactory); !ok {
			registry.ContentTypeAssociatedFactories[key] = store.NewBackingStoreSerializationWriterProxyFactory(value)
		}
	}
}

```

Reply:
Makes sense. I somehow did not really inspect the purpose of the `if` block and therefore assumed that locking outside of the for-loop is unwanted overhead if the `if` block wasn't event entered. Seems kind of silly now...

I will add that change now.

Reply:
Thanks for making the changes